ui/src/pages/backoffice/DocumentRequest/DocumentRequestDetails/components/DocumentRequestSteps/DocumentRequestSteps.js

```diff
@@ -0,0 +1,160 @@
+  Header,
+  Icon,
+  Message,
+  Segment,
+  Step,
+} from 'semantic-ui-react';
+import { BackOfficeRoutes } from '@routes/urls';
+import { documentRequest as documentRequestApi } from '@api';
+import _get from 'lodash/get';
+import { DocumentIcon } from '@pages/backoffice/components';
+import { DocumentAttachStep, DocumentProvidersStep } from './components';
+
+// This is the actual order of the steps in document request details
+const STEPS = { document: 'document', provider: 'provider', review: 'review' };
+
+export default class DocumentRequestSteps extends Component {
+  documentInfo() {
+    const { document_pid: docPid } = this.props.data.metadata;
+    const title = _get(this.props, 'data.metadata.document.title');
+    return docPid ? (
+      <Segment basic>
+        <Grid columns={2}>
+          <Grid.Column width={12}>
+            <Header>
+              Attached Document{' '}
+              <Link to={BackOfficeRoutes.documentDetailsFor(docPid)}>
+                {title}
+              </Link>
+            </Header>
+          </Grid.Column>
+          <Grid.Column width={4}>
+            <Button
+              fluid
+              name="change-document"
+              onClick={this.onChangeDocument}
+              content={'Change document'}
+            />
+          </Grid.Column>
+        </Grid>
+      </Segment>
+    ) : (
+      <Message warning>
+        <Message.Header>
+          <DocumentIcon />A document is required!
+        </Message.Header>
+        <p>
+          You can search and attach one of the existing documents, or create a
+          new one.
+        </p>
+      </Message>
+    );
+  }
+
+  renderStep(step) {
+    return (
+      <Step.Group attached="top" fluid>
+        <Step active={step === STEPS.document}>
+          <DocumentIcon />
+          <Step.Content>
+            <Step.Title>Select Document</Step.Title>
+            <Step.Description>
+              Attach a document to your request
+            </Step.Description>
+          </Step.Content>
+        </Step>
+
+        <Step
+          active={step === STEPS.provider}
+          disabled={step === STEPS.document}
+        >
+          <Icon name="truck" />
+          <Step.Content>
+            <Step.Title>Select Provider</Step.Title>
+            <Step.Description>
+              Acquire or borrow from another Library
+            </Step.Description>
+          </Step.Content>
+        </Step>
+
+        <Step active={step === STEPS.review} disabled={step !== STEPS.review}>
+          <Icon name="check" />
+          <Step.Content>
+            <Step.Title>Confirm Request</Step.Title>
+          </Step.Content>
+        </Step>
+      </Step.Group>
+    );
+  }
+
+  renderStepContent(step) {
+    const { data } = this.props;
+    switch (step) {
+      case STEPS.document:
+        return (
+          <DocumentAttachStep
+            data={data}
+            fetchDocumentRequestDetails={this.props.fetchDocumentRequestDetails}
+          />
```

Comment:
I would suggest to have two(three split components instead of having conditional props. they are different enough I think with separate goals and I think that a component should do one thing at a time (like one below, DocumentAttachStep)